### PR TITLE
ESM support

### PR DESCRIPTION
**Pull request by giltayar**
ESM support for testdouble.js, using quibble

### CODE REVIEW

**Comment by giltayar** [comment]
Note that it's _almost_ done, but not quite. Currently, to use ESM, you need to `node --loader=quibble`, but it doesn't make sense to use quibble. It should be `node --loader=testdouble`. I'll work on that (nothing major, just some glue code, and an "example" test that checks that it all hangs together), but in the meantime, if you want, you can review the code.

**Comment by searls** [comment]
I'm pretty tied up this week, so I'll wait for the all-clear that it's "ready-ready" and then do one big review/fixup en masse if that's ok

**Comment by giltayar** [comment]
Based on experience, I want to add one more thing: if you didn't specify the loader (`node --loader=testdouble`) then we should throw an exception to show you that you didn't and not fail silently with unexpected results. For this, I need to change quibble, and then come back here and do a small change.

And then I'm done. Promise! ☺️

**Comment by searls** [comment]
I shall `await` your changes!

**Comment by giltayar** [comment]
@searls And... we're done! Take your time. 

I'll start writing that blog post... :-)

File: test/safe-esm/teenytest-proxy.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env node
+
+// This file is needed because of bugs in ESM in Node.js (around loaders):
+// https://github.com/nodejs/node/issues/33226
+// https://github.com/nodejs/node/issues/33303
+//
+// Once these bugs are solved, we can go back to running teenytest regularly for ESM tests
+
+require('../../node_modules/.bin/teenytest')
```
Comment (searls) [inline test/safe-esm/teenytest-proxy.js:9]: Mind setting a todo for yourself a few months from now to come back and revisit this?

**Comment by searls** [comment]
Apologies that github is confused about the merge status, but this is merged! And released! Landed in 3.15.0

Great job seeing this through @giltayar!

**Pull request closed without merging**